Compilation errors when building with `--all-features`

# Errors
` cargo build --all-features` leads to
```
error[E0425]: cannot find value `container_id` in this scope
   --> testcontainers/src/clients/http.rs:175:49
    |
175 |             crate::watchdog::Watchdog::register(container_id.clone());
    |                                                 ^^^^^^^^^^^^ not found in this scope

error[E0599]: no variant named `DockerResponseNotFoundError` found for enum `bollard::errors::Error`
   --> testcontainers/src/clients/http.rs:151:45
    |
151 |                 Err(bollard::errors::Error::DockerResponseNotFoundError { message: _ }) => {
    |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: there is a variant with a similar name: `DockerResponseServerError`

error[E0599]: no function or associated item named `register` found for struct `Watchdog` in the current scope
   --> testcontainers/src/clients/http.rs:175:40
    |
175 |             crate::watchdog::Watchdog::register(container_id.clone());
    |                                        ^^^^^^^^ function or associated item not found in `Watchdog`
    |
   ::: testcontainers/src/watchdog.rs:33:1
    |
33  | pub(crate) struct Watchdog {
    | -------------------------- function or associated item `register` not found for this
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `register`, perhaps you need to implement it:
            candidate #1: `mio::event::source::Source`
```

# Suggested Fixes
1. `id` should be `container_id` (or the other way around): https://github.com/testcontainers/testcontainers-rs/blob/20039f83d2bf898a894fdfbd50064acb13001941/testcontainers/src/clients/http.rs#L148
2. [`bollard::errors::Error::DockerResponseNotFoundError`](https://docs.rs/bollard/0.1.0/bollard/errors/struct.DockerResponseNotFoundError.html) is now [`bollard::errors::Error::DockerResponseServerError{ status_code: 404, .. }`](https://docs.rs/bollard/latest/bollard/errors/enum.Error.html#variant.DockerResponseServerError)
3. `crate::watchdog::Watchdog::register(container_id.clone())` should be `crate::watchdog::register(container_id.clone())`: https://github.com/testcontainers/testcontainers-rs/blob/20039f83d2bf898a894fdfbd50064acb13001941/testcontainers/src/watchdog.rs#L32-L44

Maybe there should be a step in CI that builds with `--all-features` to catch errors like these.

--- a/testcontainers/src/clients/http.rs
+++ b/testcontainers/src/clients/http.rs
@@ -172,7 +172,7 @@ impl Http {
 
         #[cfg(feature = "watchdog")]
         if self.inner.command == env::Command::Remove {
-            crate::watchdog::Watchdog::register(container_id.clone());
+            crate::watchdog::register(container_id.clone());
         }
 
         self.inner

--- a/testcontainers/src/clients/http.rs
+++ b/testcontainers/src/clients/http.rs
@@ -148,7 +148,7 @@ impl Http {
         let container_id = {
             match create_result {
                 Ok(container) => container.id,
-                Err(bollard::errors::Error::DockerResponseNotFoundError { message: _ }) => {
+                Err(bollard::errors::Error::DockerResponseServerError { status_code: 404, .. }) => {
                     {
                         let pull_options = Some(CreateImageOptions {
                             from_image: image.descriptor(),

--- a/testcontainers/src/clients/http.rs
+++ b/testcontainers/src/clients/http.rs
@@ -145,7 +145,7 @@ impl Http {
         let create_result = self
             .create_container(create_options.clone(), config.clone())
             .await;
-        let id = {
+        let container_id = {
             match create_result {
                 Ok(container) => container.id,
                 Err(bollard::errors::Error::DockerResponseNotFoundError { message: _ }) => {
@@ -177,7 +177,7 @@ impl Http {
 
         self.inner
             .bollard
-            .start_container::<String>(&id, None)
+            .start_container::<String>(&container_id, None)
             .await
             .unwrap();
 
@@ -185,7 +185,7 @@ impl Http {
             inner: self.inner.clone(),
         };
 
-        ContainerAsync::new(id, client, image, self.inner.command).await
+        ContainerAsync::new(container_id, client, image, self.inner.command).await
     }
 }
 

--- a/testcontainers/src/clients/http.rs
+++ b/testcontainers/src/clients/http.rs
@@ -148,7 +148,9 @@ impl Http {
         let container_id = {
             match create_result {
                 Ok(container) => container.id,
-                Err(bollard::errors::Error::DockerResponseServerError { status_code: 404, .. }) => {
+                Err(bollard::errors::Error::DockerResponseServerError {
+                    status_code: 404, ..
+                }) => {
                     {
                         let pull_options = Some(CreateImageOptions {
                             from_image: image.descriptor(),

